### PR TITLE
Improve quantile method docstring + error

### LINKED ISSUE #6875: update numpy version requirement
### What happened?

in the latest xarray version 2022.06.0, xarray.DataArray.quantile supports 9 methods:
https://docs.xarray.dev/en/latest/generated/xarray.DataArray.quantile.html

however, those methods are only implemented starting from numpy  1.22.0 
https://numpy.org/doc/stable/reference/generated/numpy.quantile.html

whereas the current numpy requirement is only numpy >= 1.19

### What did you expect to happen?

that the numpy requirement is updated to numpy >= 1.22

### Minimal Complete Verifiable Example

_No response_

### MVCE confirmation

- [ ] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [ ] New issue — a search of GitHub Issues suggests this is not a duplicate.

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>

</details>

### PR DESCRIPTION
**Pull request by headtr1ck**
- [x] Closes #6875
- [x] ~~Tests added~~
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] ~~New functions/methods are listed in `api.rst`~~

### CODE REVIEW

File: xarray/core/variable.py [hunk 1]
```diff
@@ -2140,6 +2141,10 @@ def _wrapper(npa, **kwargs):
         if Version(np.__version__) >= Version("1.22.0"):
             kwargs = {"q": q, "axis": axis, "method": method}
         else:
+            if method not in ("linear", "lower", "higher", "midpoint", "nearest"):
+                raise ValueError(
+                    "Interpolation method '{method}' requires numpy >= 1.22 or is not supported."
```
Comment (mathause) [inline xarray/core/variable.py:2146]:
```suggestion
                    f"Interpolation method '{method}' requires numpy >= 1.22 or is not supported."
```

File: setup.cfg [hunk 2]
```diff
@@ -75,7 +75,7 @@ zip_safe = False  # https://mypy.readthedocs.io/en/latest/installed_packages.htm
 include_package_data = True
 python_requires = >=3.8
 install_requires =
-    numpy >= 1.19
+    numpy >= 1.19  # recommended to use >= 1.22 for full quantile method support
```
Comment (mathause) [inline setup.cfg:78]: I am slightly wary of this - I guess there is a lot of details that are not working on older supported versions. But ok to leave it.

**Review by mathause — APPROVED** [review]
Forgotten `f""` but looks good otherwise.

File: setup.cfg (hunk 2 above)
Comment (dcherian) [reply setup.cfg:78]: yeah me too. this is just another thing to keep updating for what is a standard assumption in the python ecosystem.

Comment (headtr1ck) [reply setup.cfg:78]: Yeah. It was just specifically requested in the issue.
Feel free to propose a more general comment.

Comment (mathause) [reply setup.cfg:78]: How about
```
# minimum supported versions - install the newest version for best support
```
?

Comment (headtr1ck) [reply setup.cfg:78]: Let's ping the person that created the issue https://github.com/smoors.
I never look into the setup file when installing packages, so better to ask someone who does?

Comment (smoors) [reply setup.cfg:78]: thanks a lot for doing this!

I actually prefer the original comment with the specific minimum version and reason why. without that info, it's basically useless.

also, don't think it's a maintenance burden at all. once the numpy requirement is bumped to >= 1.22 or higher, you can simply remove the comment.

but I'm not an `xarray` developer, so ... :)

Comment (mathause) [reply setup.cfg:78]: It's not this comment per se, but the expectation to list the compatibilities in setup.py. But anyhow - I suggest to merge as is with the disclaimer that this is not something we generally plan to do. Objections?

**Comment by mathause** [comment]
Thanks @headtr1ck!

**Pull request merged**